[c10d] Update `backend` arg documentation

kwen2501 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #142404

Update doc to reflect change brought by https://github.com/pytorch/pytorch/pull/142216

cc @H-Huang @awgu @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

kwen2501 committed:
[c10d] Update `backend` arg documentation

[ghstack-poisoned]

XilunWu commented on torch/distributed/distributed_c10d.py:
```diff
@@ -1542,15 +1542,22 @@ def init_process_group(
+            for the device type indicated by the `device_id` kwarg (if
+            provided). The known default registrations today are: ``nccl`` for
+            ``cuda``, ``gloo`` for ``cpu``.
+            If neither ``backend`` nor ``device_id`` is provided, c10d will
+            detect the accelerator on the run-time machine and use a backend
+            known for that detected accelerator (or ``cpu``).
```
didn't we require users to specify ``backend`` if an accelerator other than Nvidia GPU is detected?

kwen2501 replied:
We will error out and ask for the `backend` field if we do not know which backend to use. This is general.

kwen2501 committed:
Update on "[c10d] Update `backend` arg documentation"


Update doc to reflect change brought by https://github.com/pytorch/pytorch/pull/142216

cc H-Huang awgu wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]

kwen2501 committed:
Update on "[c10d] Update `backend` arg documentation"


Update doc to reflect change brought by https://github.com/pytorch/pytorch/pull/142216

cc H-Huang awgu wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]

kwen2501 commented:
@pytorchbot merge
